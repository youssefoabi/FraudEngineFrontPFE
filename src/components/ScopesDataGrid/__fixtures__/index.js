export default [
  {
    id: 1,
    name: 'string1',
    urlTemplate: 'stringC',
    modifiedAt: '1586524639323',
  },
  {
    id: 2,
    name: 'string2',
    urlTemplate: 'stringA',
    modifiedAt: '1586611039323',
  },
  {
    id: 3,
    name: 'string3',
    urlTemplate: 'stringB',
    modifiedAt: '1586438239323',
  },
];
