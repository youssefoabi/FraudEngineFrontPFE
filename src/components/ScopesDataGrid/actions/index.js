export const SET_SCOPES_PAGE = '@SCOPES/SET_PAGE';
export const SET_SCOPES_ROWS_PER_PAGE = '@SCOPES/SET_ROWS_PER_PAGE';
export const SET_SCOPES_SORT = '@SCOPES/SET_SORT';
export const SET_SCOPES_SELECTED = '@SCOPES/SET_SELECTED';
export const SET_SCOPES = '@SCOPES/SET';
export const FETCH_SCOPES = '@SCOPES/FETCH';
export const DELETE_SCOPES_ERROR = '@SCOPES/DELETE_SCOPES_FAILURE';
export const DELETE_SCOPES_REQUEST = '@SCOPES/DELETE_SCOPES_REQUEST';
export const DELETE_SCOPES_SUCCESS = '@SCOPES/DELETE_SCOPES_SUCCESS';
export const RESET_DELETE_SCOPES_ERROR = '@SCOPES/RESET_DELETE_SCOPES_ERROR';
export const RESET_DELETE_SCOPES_SUCCESS = '@SCOPES/RESET_DELETE_SCOPES_SUCCESS';
