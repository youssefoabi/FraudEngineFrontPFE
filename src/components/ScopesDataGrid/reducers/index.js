export { default as scopesNavigation } from './navigationReducer';
export { default as scopesEntities } from './entitiesReducer';
