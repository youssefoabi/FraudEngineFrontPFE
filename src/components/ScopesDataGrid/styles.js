export default function style() {
  return {
    inlineText: {
      fontSize: 'inherit',
      whiteSpace: 'nowrap',
    },
  };
}
