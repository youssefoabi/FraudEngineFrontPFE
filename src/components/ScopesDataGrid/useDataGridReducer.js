import { path } from 'ramda';
import { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { SET_SCOPES_PAGE, SET_SCOPES_ROWS_PER_PAGE, SET_SCOPES_SELECTED, SET_SCOPES_SORT } from './actions';

export default function useDataGridReducer() {
  const page = useSelector(path(['navigation', 'scopes', 'page']));
  const rowsPerPage = useSelector(path(['navigation', 'scopes', 'rowsPerPage']));
  const selected = useSelector(path(['navigation', 'scopes', 'selected']));
  const order = useSelector(path(['navigation', 'scopes', 'order']));
  const orderBy = useSelector(path(['navigation', 'scopes', 'orderBy']));

  const dispatch = useDispatch();
  const setPage = useCallback(nextPage => dispatch({ type: SET_SCOPES_PAGE, payload: nextPage }), [dispatch]);
  const setSort = useCallback(nextSort => dispatch({ type: SET_SCOPES_SORT, payload: nextSort }), [dispatch]);
  const setRowsPerPage = useCallback(
    nextRowsPerPage => dispatch({ type: SET_SCOPES_ROWS_PER_PAGE, payload: nextRowsPerPage }),
    [dispatch],
  );
  const setSelected = useCallback(
    nextSelected => dispatch({ type: SET_SCOPES_SELECTED, payload: nextSelected }),
    [dispatch],
  );

  return { page, rowsPerPage, order, orderBy, selected, setPage, setRowsPerPage, setSort, setSelected };
}
