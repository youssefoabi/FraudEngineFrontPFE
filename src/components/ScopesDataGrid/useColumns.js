import { Typography } from '@material-ui/core';
import { useStyles } from '@cdiscount/backoffice-ui';
import format from 'date-fns/format';
import React from 'react';

import styles from './styles';

export default function useColumns() {
  const classes = useStyles(styles);

  function renderInlineCell(text) {
    return <Typography className={classes.inlineText}>{text}</Typography>;
  }

  function renderInlineDateCell(date) {
    return date && <Typography className={classes.inlineText}>{format(date, 'dd/MM/yyyy HH:mm')}</Typography>;
  }

  const columns = [
    {
      id: 'name',
      label: 'Gamme',
      render: renderInlineCell,
      isSortable: true,
    },
    {
      id: 'urlTemplate',
      label: 'URL',
      render: renderInlineCell,
      isSortable: true,
    },
    {
      id: 'modifiedAt',
      label: 'Derni√®re modification',
      render: renderInlineDateCell,
      isSortable: true,
    },
  ];

  return { columns };
}
