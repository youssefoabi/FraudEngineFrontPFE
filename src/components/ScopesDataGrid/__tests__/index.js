import { itRendersAllMutations } from '@cdiscount/jest-utils';

import ScopesDataGrid from '../';

jest.mock('../useColumns', () => () => ({
  scopes: [],
}));

jest.mock('use-react-router', () => () => ({ location: {}, history: { push: jest.fn() } }));

jest.mock('../useDataGridReducer', () => () => ({
  page: 2,
  rowsPerPage: 10,
  order: 'asc',
  orderBy: null,
  selected: [],
}));

describe(ScopesDataGrid.name, () => {
  itRendersAllMutations(ScopesDataGrid, [{ name: 'without props', props: {} }]);
});
