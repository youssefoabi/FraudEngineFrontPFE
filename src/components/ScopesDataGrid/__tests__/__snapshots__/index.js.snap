// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid DataGrid renders correctly without props 1`] = `
<S
  actions={Array []}
  align="center"
  bodyActions={
    Array [
      Object {
        "icon": Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "displayName": "EditIcon",
          "muiName": "SvgIcon",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "label": "Modifier",
        "onClick": [Function],
      },
    ]
  }
  bottomPagination={false}
  checkbox={true}
  columns={null}
  isLoading={false}
  labelDisplayedRows={null}
  labelRowsPerPage={null}
  onChangePage={[Function]}
  onChangeRowsPerPage={[Function]}
  onRequestSort={[Function]}
  onSelect={[Function]}
  onSelectAll={[Function]}
  order="asc"
  orderBy={null}
  page={2}
  placeholderLabel="No records. 🕳️"
  rowCount={0}
  rows={Array []}
  rowsPerPage={10}
  rowsPerPageOptions={
    Array [
      10,
      15,
      20,
    ]
  }
  selected={Array []}
  square={false}
/>
`;
